src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -28,20 +28,37 @@ public JsonConsoleFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options
 
         public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider? scopeProvider, TextWriter textWriter)
         {
-            string message = logEntry.Formatter(logEntry.State, logEntry.Exception);
-            if (logEntry.Exception == null && message == null)
+            if (logEntry.State is BufferedLogRecord bufferedRecord)
             {
-                return;
+                string message = bufferedRecord.FormattedMessage ?? string.Empty;
+                if (bufferedRecord.Exception == null && message == null)
+                {
+                    return;
+                }
+
+                WriteInternal(scopeProvider, textWriter, message, bufferedRecord.LogLevel, logEntry.Category, bufferedRecord.EventId.Id, bufferedRecord.Exception,
```

Comment:
>scopeProvider [](http://example.com/codeflow?start=30&length=13)

I think the scope provider shouldn't be used here as I expect the generated formatted message already include the scope info.